common/src/main/java/com/microsoft/identity/common/internal/util/AndroidKeyStoreUtil.java

```diff
@@ -447,11 +449,27 @@ public static synchronized SecretKey unwrap(@NonNull final byte[] wrappedKeyBlob
                 exception
         );
         if (exception instanceof InvalidKeyException) {
+            final android.security.KeyStoreException keyStoreException = findKeyStoreException(exception);
+            String ksMessage;
+            final String ksNumericErrorCode;
+            if (keyStoreException != null) {
+                ksMessage = keyStoreException.getMessage();
+                if (ksMessage == null) {
+                    ksMessage = "Keystore exception found, no error message";
+                }
+                ksNumericErrorCode = getNumericErrorCodeFromKeyStoreException(keyStoreException);
+            } else {
+                ksMessage = "No keystore exception found";
+                ksNumericErrorCode = "";
+            }
+
             final Attributes attributes = Attributes.builder()
                     .put(AttributeName.keystore_operation.name(), "unwrap")
                     .put(AttributeName.error_code.name(), errCode)
                     .put(AttributeName.error_type.name(), clientException.getClass().getSimpleName())
```

Comment:
>clientException.getClass().getSimpleName() [](http://example.com/codeflow?start=58&length=42)

this is always ClientExeption. we rather replace it with excception.getMessage(), here exception = InvalidKeyEception. this can be helpful if InvalidKeyException is not caused by KeyStoreException. You will have some info.